### PR TITLE
adjust example to be more aligned with python-local analytics

### PR DESCRIPTION
**Pull request by mccanne**

### CODE REVIEW

File: rayzardemo.py [hunk 1]
```diff
@@ -97,12 +104,35 @@ def textzng(zng):
...
-    srcs = ray.get(zar.find.remote(args.find))
-    xformed = [zq.remote(args.filter, src) for src in srcs]
+    # Launch a ray actor to search for the log chunks that match the query.
+    search = ArchiveSearch.remote(args.zarroot, args.find)
 
+    # Launch a ray actor to aggregate all of the matching chunks.
     agg = Aggregator.remote()
-    res = ray.get(agg.aggregate.remote(args.merge, xformed))
+
+    maxtasks = 4 # XXX
```
Comment (alfred-landrum) [inline rayzardemo.py]: This value could come from a cli flag.

File: rayzardemo.py [hunk 2]
```diff
@@ -42,37 +38,48 @@ def load(location, filter=None):
...
-@ray.remote
-def zqcombine(zql, obj_ids):
-    zng = bytearray()
-    for o in obj_ids:
-        zng.extend(ray.get(o))
-    return localzq(zql, zng)
-
+        self.locs = p.stdout.splitlines()
+
+    @ray.method(num_return_vals=1)
```
Comment (alfred-landrum) [inline rayzardemo.py]: I believe the `ray.method` annotation is only necessary if you need to specify that the number of object ids returned by the method will be something other than 1.

File: rayzardemo.py [hunk 3]
```diff
@@ -97,12 +104,35 @@ def textzng(zng):
...
+    tasks = []
+
+    # Loop over the search results and do an parallel aggregation up to
+    # maxtasks in parallel.
+    while True:
+        location = ray.get(search.next.remote())
+        if location == None:
+            break
+        # For each hit, we launch an actor to read the chunk and filter
+        # to only the events that match the search.
```
Comment (alfred-landrum) [inline rayzardemo.py]:
```suggestion
        # For each hit, we create a task to read the chunk and filter
        # to only the events that match the search.
```

**Review by alfred-landrum — APPROVED** [review]

File: rayzardemo.py (hunk 3 above)
Comment (mccanne) [reply rayzardemo.py]: But it's creating a Ray actor, which is stateful, not a stateless Ray task... ?

File: rayzardemo.py (hunk 2 above)
Comment (mccanne) [reply rayzardemo.py]: That is correct.  Will fix.

File: rayzardemo.py (hunk 1 above)
Comment (mccanne) [reply rayzardemo.py]: ok

File: rayzardemo.py (hunk 3 above)
Comment (alfred-landrum) [reply rayzardemo.py]: My understanding of ray is obviously limited so far, but my interpretation of their docs is that the line above where you perform `agg = Aggregator.remote()` is that it represents 'launching an actor'; however, here you are calling a method on that existing actor, not launching a new one?

Comment (mccanne) [reply rayzardemo.py]: I stand corrected.  It is a stateless loading of the file the corresponds each generation of a path from the actor that is searching.  I'll fix the comment, thanks!

**Pull request merged**